### PR TITLE
Add Freemuxlet to modules

### LINKED ISSUE #5073: new module: demuxlet/freemuxlet
### Is there an existing module for this?

- [X] I have searched for the existing module

### Is there an open PR for this?

- [X] I have searched for existing PRs

### Is there an open issue for this?

- [X] I have searched for existing issues

### Are you going to work on this?

- [x] If I'm planning to work on this module, I added myself to the `Assignees` to facilitate tracking who is working on the module

### PR DESCRIPTION
**Pull request by wxicu**
## PR checklist
Add freemuxlet to modules.
Closes #5073 

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - For modules:
    - [ ] `nf-core modules test <MODULE> --profile docker`
    - [ ] `nf-core modules test <MODULE> --profile singularity`
    - [x] `nf-core modules test <MODULE> --profile conda`
  - For subworkflows:
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile docker`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile singularity`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile conda`

### CODE REVIEW

File: modules/nf-core/popscle/freemuxlet/meta.yml [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+  - singlet_vcf:
+      type: file
+      description: Output vcf file for each sample inferred and clustered from freemuxlet assuming all droplets are singlets.
+      pattern: "*.clust0.vcf.gz"
+
+  - ldist:
+      type: file
+      description: Output ontains the pairwise Bayes factor for each possible pair of droplets.
+      pattern: "*.ldist.gz"
```
Comment (nvnieuwk) [inline modules/nf-core/popscle/freemuxlet/meta.yml:73]: Can you add to the description how these optional files can be created?

File: modules/nf-core/popscle/freemuxlet/tests/main.nf.test
Comment (nvnieuwk) [inline modules/nf-core/popscle/freemuxlet/tests/main.nf.test:1]: Can you add a test that also outputs the optional files?

File: modules/nf-core/popscle/freemuxlet/main.nf [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    """
+    popscle freemuxlet \\
+        --plp ${plp}/$prefix \\
+        --out $prefix \\
+        --nsample $n_sample \\
+        $args
+
+    cat <<-END_VERSIONS > versions.yml
+    "${task.process}":
+        popscle freemuxlet: $VERSION
```
Comment (nvnieuwk) [inline modules/nf-core/popscle/freemuxlet/main.nf:38]:
```suggestion
        popscle: $VERSION
```

File: modules/nf-core/popscle/freemuxlet/main.nf [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    def VERSION = '0.1' // WARN: Version information not provided by tool on CLI. Please update version string below when bumping container versions.
+
+    """
+    touch ${prefix}.clust1.samples.gz
+    touch ${prefix}.clust1.vcf.gz
+    touch ${prefix}.lmix
+
+    cat <<-END_VERSIONS > versions.yml
+    "${task.process}":
+        popscle freemuxlet: $VERSION
```
Comment (nvnieuwk) [inline modules/nf-core/popscle/freemuxlet/main.nf:54]:
```suggestion
        popscle: $VERSION
```

File: modules/nf-core/popscle/freemuxlet/main.nf [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+    stub:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    def VERSION = '0.1' // WARN: Version information not provided by tool on CLI. Please update version string below when bumping container versions.
+
+    """
+    touch ${prefix}.clust1.samples.gz
+    touch ${prefix}.clust1.vcf.gz
+    touch ${prefix}.lmix
```
Comment (nvnieuwk) [inline modules/nf-core/popscle/freemuxlet/main.nf:50]: Can you add the optional creation of the optional files to the stub? These should be created when the right arguments are used

**Review by nvnieuwk — CHANGES_REQUESTED** [review]
Hi, thank you for your contribution! Here are some suggestions and comments :)

File: modules/nf-core/popscle/freemuxlet/main.nf [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+
+    if [[ "$args" == *"--aux-files"* ]]; then
+        touch ${prefix}.clust0.samples.gz
+        touch ${prefix}.clust0.vcf.gz
+    fi
```
Comment (nvnieuwk) [inline modules/nf-core/popscle/freemuxlet/main.nf:54]: will this also work if more arguments than `--aux-files` have been given?

Comment (wxicu) [reply modules/nf-core/popscle/freemuxlet/main.nf:54]: yes it works as long as it contains `--aux-files`

**Review by nvnieuwk — APPROVED** [review]

**Pull request merged**